libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -905,12 +915,47 @@ internal class MapRouteLine(
         }
     }
 
+    /**
+     * Updates the route line appearance from the origin point to the point indicated in
+     * @param point representing the portion of the route that has been traveled.
+     */
+    fun updateTraveledRouteLine(point: Point) {
+        ifNonNull(primaryRoute, primaryRoute?.distance()) { primaryRoute, routeDistance ->
+            val lineString = getLineStringForRoute(primaryRoute)
+            val nearestPointOnLineDistance = TurfMisc.nearestPointOnLine(
+                point,
+                lineString.coordinates(),
+                TurfConstants.UNIT_METERS
+            ).getProperty("dist").asDouble
+
+            if (
+                nearestPointOnLineDistance > ROUTE_LINE_UPDATE_MAX_DISTANCE_THRESHOLD_IN_METERS ||
+                vanishingPointUpdateInhibited
+            ) {
+                return
+            }
+
+            val distanceTraveled = MapRouteLineSupport.findDistanceOfPointAlongLine(
+                lineString,
+                point
```

Comment:
We should use the `nearestPointOnLine` here, shouldn't we?

Reply:
NearestPointOnLine doesn't tell me what I need to know. If the route overlaps I won't know where the returned point is on the route line.  I need the distance traveled. In order to get that I have to get to the nearest point by processing the points along the route.

Reply:
Sure, but the point of reference you're calculating against should be the `nearestPointOnLine`, not the potentially not-map-matched point.

Reply:
The nearest point returned by Turf is nearest in a straight line not the nearest traveling along the route. That's what I'm calculating here.  If you look at line 929 there's another calculation to get the nearest point on the line which is used to check if the map matching is off by some threshold.